File: 350_intersection_of_two_arrays_II/intersectSort.py
from typing import List


class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        a = sorted(nums1)
        b = sorted(nums2)
        res = []
        i = j = 0

        while i < len(a) and j < len(b):
            if a[i] == b[j]:
                res.append(a[i])
                i += 1
                j += 1

            elif a[i] < b[j]:
                i += 1
            else:
                j += 1

        return res

File: 350_intersection_of_two_arrays_II/intersectSort.go
package intersectionoftwoarraysii

import "sort"

func intersect(nums1 []int, nums2 []int) []int {
	sort.Ints(nums1)
	sort.Ints(nums2)

	var res []int

	i, j := 0, 0
	n, m := len(nums1), len(nums2)

	for i < n && j < m {
		if nums1[i] == nums2[j] {
			res = append(res, nums1[i])
			i++
			j++
		} else if nums1[i] < nums2[j] {
			i++
		} else {
			j++
		}
	}

	return res
}